### PR TITLE
fixed delete-image error code

### PR DESCRIPTION
**Pull request by tomershinhar**
# Description

when trying to delete an image with status "SUCCESS", the rc should be 400

Fixes # (issue)
[THEEDGE-2863](https://issues.redhat.com/browse/THEEDGE-2863)

## Type of change

What is it?

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

File: pkg/routes/images.go [hunk 1]
```diff
@@ -767,7 +767,7 @@ func DeleteImage(w http.ResponseWriter, r *http.Request) {
 
 	err := ctxServices.ImageService.DeleteImage(image)
 	if err != nil {
-		err := errors.NewInternalServerError()
+		err := errors.NewBadRequest("Given image can't be deleted")
```
Comment (ldjebran) [inline pkg/routes/images.go:770]: I believe that this should a switch when ImageNotInErrorState we return 400 , and when other we return 500
something like this https://github.com/RedHatInsights/edge-api/blob/main/pkg/routes/devicegroups.go#L89

**Review by omaciel — DISMISSED** [review]
Assuming @ldjebran suggestion is implemented, lgtm

File: pkg/routes/images.go (hunk 1 above)
Comment (tomershinhar) [reply pkg/routes/images.go]: @ldjebran done,
can you please take a look?

File: pkg/routes/images.go [hunk 2]
```diff
@@ -767,9 +767,16 @@ func DeleteImage(w http.ResponseWriter, r *http.Request) {
 
 	err := ctxServices.ImageService.DeleteImage(image)
 	if err != nil {
-		err := errors.NewInternalServerError()
-		err.SetTitle("Failed deleting image")
-		respondWithAPIError(w, ctxServices.Log, err)
+		var responseErr errors.APIError
+		switch err.(type) {
+		case *services.ImageNotInErrorState:
+			responseErr = errors.NewBadRequest("Given image can't be deleted")
```
Comment (mgold1234) [inline pkg/routes/images.go:773]: is it possible to add a reason why we cant delete this image?

**Review by mgold1234 — APPROVED** [review]
/lgtm

**Comment by lahavyuv86** [comment]
LGTM

**Review by lahavyuv86 — APPROVED** [review]

**Pull request merged**